ElrondNetwork/elrond-go
# EN-13476: Make esdt supply endpoint work for collections

## Reasoning behind the pull request

- Currently, the esdt supply endpoint only works for a nonce of an SFT/MetaESDT. for example, you can know the total supply of LKMEX-aab910-01, but you cannot know the supply of LKMEX-aab910.
  
## Proposed changes

- Extend the esdtSuppliesProcessor to also save the total amount for a collection. For example, when there is a local mint for LKMEX-aab910-01, reflect the same change on the LKMEX-aab910 as well 

## Testing procedure
- Issue a non-fungible token (ABC-123456)
- Mint some tokens with nonce 1 (ABC-123456-01)
- Mint some tokens with nonce 2 (ABC-123456-02)
- Check the correct supply & minted values for both nonce [1](esdt/supply/ABC-123456-01) and [2](esdt/supply/ABC-123456-02)
- Check the correct supply & minted for the [whole collection](esdt/supply/ABC-123456) (the values are the sum for each nonce of the token) 
- Burn some tokens with nonce 1
- Burn some tokens with nonce 2
- Check the correct burned values for both nonce [1](esdt/supply/ABC-123456-01) and [2](esdt/supply/ABC-123456-02)
- Check the correct burned for the [whole collection](esdt/supply/ABC-123456) (the values are the sum for each nonce of the token)

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
